controllers/networking-calico/pkg/charts/utils.go

```diff
@@ -27,68 +30,167 @@ const (
 	usePodCIDR = "usePodCidr"
 )
 
+type calicoConfig struct {
```

Comment:
This was introduced to get rid of the map[string]interface handling below

Reply:
I am not sure if it is really needed, point is we this struct which seems like duplicate code somehow, then we convert it back and forth, at the end we have to parse it as `map[string]interface` anyways, I was just asking if there is some other use-case or need that prompted such change which I might be missing. 

I don't really mind it, I was just asking what was the reasoning to do so :)

Reply:
We had the same discussion, but because in utils.go there where already structs for calico configuration we decided to extend them. But puttings these into types.go is way cleaner, agreed. I talk to my colleagues and come back.

Reply:
I think, in the beginning we tried to keep all mechanics the way they were and did not want to introduce so many changes. We worked with the `map[string]interface`s, but it turned out that nested structs are not as nice to write and due to the missing type-safety it was easier to make an error. Now we can express the default chart configuration like this, which I think is good to read and to understand:

```
var defaultCalicoConfig = calicoConfig{
	Backend: calicov1alpha1.Bird,
	Felix: felix{
		IPInIP: felixIPinIP{
			Enabled: true,
		},
	},
	IPv4: ipv4{
		Pool:                calicov1alpha1.PoolIPIP,
		Mode:                calicov1alpha1.Always,
		AutoDetectionMethod: nil,
	},
	IPAM: ipam{
		IPAMType: hostLocal,
		Subnet:   usePodCIDR,
	},
	Typha: typha{
		Enabled: true,
	},
}
```

In my opinion it is ok that we have two different configuration structs, one representing the API representation the other one the chart representation.

Reply:
makes sense, thanks for clarifying :)